v19.2/migrate-from-oracle.md

```diff
@@ -0,0 +1,400 @@
+{% include copy-clipboard.html %}
+~~~ shell
+$ impdp user/password directory=datapump dumpfile=oracle_example.dmp sqlfile=example_sql.sql TRANSFORM=SEGMENT_ATTRIBUTES:N:table PARTITION_OPTIONS=MERGE
+~~~
+
+This SQL output will be used later, in [Step 7](#step-7-map-oracle-to-cockroachdb-data-types).
+
+## Step 3. Export table data
+
+You need to extract each table's data into a data list file (`.lst`). We wrote a simple SQL script(`spool.sql`) to do this:
+
+~~~ shell
+$ cat spool.sql
+
+SET ECHO OFF
+SET TERMOUT OFF
+SET FEEDBACK OFF
+SET PAGESIZE 0
+SET TRIMSPOOL ON
+SET WRAP OFF
+set linesize 30000
+SET RECSEP OFF
+SET VERIFY OFF
+SET ARRAYSIZE 10000
+SET COLSEP '|'
+
+SPOOL '&1'
+
+ALTER SESSION SET nls_date_format = 'YYYY-MM-DD HH24:MI:SS';
+  # Used to set a properly formatted date for CockroachDB
+
+SELECT * from &1;
+
+SPOOL OFF
+
+SET PAGESIZE 24
+SET FEEDBACK ON
+SET TERMOUT ON
+~~~
+
+To extract the data, we ran the script for each table in SQL*Plus:
+
+{% include copy-clipboard.html %}
+~~~ sql
+$ sqlplus user/password
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> @spool CUSTOMERS
+  @spool ADDRESSES
+  @spool CARD_DETAILS
+  @spool WAREHOUSES
+  @spool ORDER_ITEMS
+  @spool ORDERS
+  @spool INVENTORIES
+  @spool PRODUCT_INFORMATION
+  @spool LOGON
+  @spool PRODUCT_DESCRIPTIONS
+  @spool ORDERENTRY_METADATA
+~~~
+
+A data list file (`.lst`) with leading and trailing spaces is created for each table.
+
+Exit SQL*Plus:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> EXIT
+~~~
+
+## Step 4. Configure and convert the table data to CSV
+
+Each table's data list file needs to be converted to CSV and formatted for CockroachDB. We wrote a simple Python script (`fix-example.py`) to do this:
+
+~~~ python
+$ cat fix-example.py
+
+import csv
+import string
+import sys
+
+for lstfile in sys.argv[1:]:
+  filename = lstfile.split(".")[0]
+
+  with open(sys.argv[1]) as f:
+    reader = csv.reader(f, delimiter="|")
+    with open(filename+".csv", "w") as fo:
+      writer = csv.writer(fo, delimiter="|")
+      for rec in reader:
+        writer.writerow(map(string.strip, rec))
+~~~
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ python2 fix.py CUSTOMERS.lst ADDRESSES.lst CARD_DETAILS.lst WAREHOUSES.lst ORDER_ITEMS.lst ORDERS.lst INVENTORIES.lst PRODUCT_INFORMATION.lst LOGON.lst PRODUCT_DESCRIPTIONS.lst ORDERENTRY_METADATA.lst
+~~~
+
+Format the generated CSV files to meet the CockroachDB's [CSV requirements](#csv-requirements).
+
+### CSV requirements
+
+You will need to export one CSV file per table, with the following requirements:
+
+- Files must be in [valid CSV format](https://tools.ietf.org/html/rfc4180), with the caveat that the delimiter must be a single character.  To use a character other than comma (such as a tab), set a custom delimiter using the [`delimiter` option](import.html#delimiter).
+- Files must be UTF-8 encoded.
+- If one of the following characters appears in a field, the field must be enclosed by double quotes:
```

Comment:
nit: Should "double quote" be "double-quote"? If yes, change throughout the doc.